gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java

```diff
@@ -0,0 +1,342 @@
+/* (c) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.source.extractor.extract.kafka;
+
+import gobblin.configuration.State;
+import gobblin.source.extractor.extract.MessageBasedSource;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import kafka.api.PartitionOffsetRequestInfo;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.OffsetRequest;
+import kafka.javaapi.OffsetResponse;
+import kafka.javaapi.PartitionMetadata;
+import kafka.javaapi.TopicMetadata;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.consumer.SimpleConsumer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import parquet.Preconditions;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+
+/**
+ * Wrapper class that contains two alternative Kakfa APIs: an old low-level Scala-based API, and a new API.
+ * The new API has not been implemented since it's not ready to be open sourced.
+ *
+ * @author ziliu
+ *
+ */
+public class KafkaWrapper implements Closeable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaWrapper.class);
+
+  private static final String KAFKA_API_TYPE = "kafka.api.type";
+  private static final APIType DEFAULT_KAFKA_API_TYPE = APIType.OLD;
+  private static final String KAFKA_BROKERS = "kafka.brokers";
+
+  private final APIType apiType;
+  private final List<String> brokers;
+  private final KafkaAPI kafkaAPI;
+
+  private enum APIType {
```

Comment:
If there are only the old and new APIs, you can use a boolean here instead of this enum.